lib/bolt/core/src/context/project.rs

```diff
@@ -234,6 +234,14 @@ impl ProjectContextData {
 		{
 			panic!("must have host networking enabled if tests + pools are enabled (rivet.matchmaker.host_networking = true)");
 		}
+
+		// MARK: Billing emails
+		if self.ns().rivet.billing.is_some() {
+			assert!(
+				self.ns().email.is_some(),
+				"cannot enable billing without emailing"
+			);
+		}
```

Comment:
I know there's many ways of implementing the boolean logic here, do you have any standard we should go by?

Reply:
not sure i understand the q. like why not use `assert!(billing.is_some() && email.is_some())` (that's different logic than what's impl).

Reply:
in this specific case you could do either of these

```
assert(!a || b)
if a { if !b { panic!() } }
if a { assert!(b) }
```

Reply:
ah yeah. i kinda like this method better since it helps separate out the logic a bit better. ie the dedicated if helps state "this is going to be all assertions related to billing." definitely nit-picking at this point.